pkg/grafana/datasource-handler.go

```diff
@@ -133,3 +136,28 @@ func (h *DatasourceHandler) Add(resource grizzly.Resource) error {
 func (h *DatasourceHandler) Update(existing, resource grizzly.Resource) error {
 	return putDatasource(resource)
 }
+
+func (h *DatasourceHandler) getRemoteDatasource(uid string) (*grizzly.Resource, error) {
+	ds, err := h.Provider.client.DataSourceByUID(uid)
+
+	var nf gapi.ErrNotFound
+	if err != nil && errors.As(err, &nf) {
+		ds, err = h.Provider.client.DataSourceByUID(uid) // TODO: Replace by DataSourceByName
```

Comment:
Waiting for https://github.com/grafana/grafana-api-golang-client/pull/164 to be merged